core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -373,11 +377,255 @@ User revokeRolesFromUser(String metalake, List<String> roles, String user) {
+                                                    role, oldObject, newSecurableObject));
+                                          }));
+
+                              return newSecurableObject;
+                            } else {
+                              // If the object doesn't contain any privilege, we remove this object.
+                              authorizationCallbackWrapper.setCallBack(
+                                  () ->
+                                      AuthorizationUtils.callAuthorizationPluginForMetadataObject(
+                                          metalake,
+                                          object,
+                                          authorizationPlugin -> {
+                                            authorizationPlugin.onRoleUpdated(
+                                                roleEntity,
+                                                RoleChange.removeSecurableObject(role, oldObject));
+                                          }));
+
+                              return null;
+                            }
+                          }
+                        });
+
+                AuditInfo auditInfo =
+                    AuditInfo.builder()
+                        .withCreator(roleEntity.auditInfo().creator())
+                        .withCreateTime(roleEntity.auditInfo().createTime())
+                        .withLastModifier(PrincipalUtils.getCurrentPrincipal().getName())
+                        .withLastModifiedTime(Instant.now())
+                        .build();
+
+                return RoleEntity.builder()
+                    .withId(roleEntity.id())
+                    .withName(roleEntity.name())
+                    .withNamespace(roleEntity.namespace())
+                    .withProperties(roleEntity.properties())
+                    .withAuditInfo(auditInfo)
+                    .withSecurableObjects(updateSecurableObjects)
+                    .build();
+              });
+
+      authorizationCallbackWrapper.execute();
+
+      return updatedRole;
+    } catch (NoSuchEntityException nse) {
+      LOG.error("Failed to revoke, role {} does not exist in the metalake {}", role, metalake, nse);
+      throw new NoSuchRoleException(ROLE_DOES_NOT_EXIST_MSG, role, metalake);
+    } catch (IOException ioe) {
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  private List<SecurableObject> updateSecurableObjects(
+      List<SecurableObject> securableObjects,
+      MetadataObject targetObject,
+      Function<SecurableObject, SecurableObject> objectUpdater) {
+    boolean isExist = false;
+    List<SecurableObject> updateSecurableObjects = Lists.newArrayList();
+    for (SecurableObject securableObject : securableObjects) {
+      // If securable object is matching the target object, we apply the updates
+      // to this securable object.
+      if (!isExist
+          && securableObject.fullName().equals(targetObject.fullName())
```

Comment:
I wonder if the securableObject has method like `equals`?

Reply:
If it support equals, I think we can use a set to reduce the logic.

Reply:
This compares a metadata object and a securable object.

Reply:
I replace with equals.

Reply:
After discussing, this option can be ignored.